packages/repository/src/decorators/model.decorator.ts

```diff
@@ -91,6 +92,16 @@ export function model(definition?: Partial<ModelDefinitionSyntax>) {
  * @returns {(target:any, key:string)}
  */
 export function property(definition?: Partial<PropertyDefinition>) {
+  if (
+    definition &&
+    (definition.type === Array || definition.type === 'array') &&
+    (definition as object).hasOwnProperty('itemType') &&
```

Comment:
this logic here only accounts for circular dependency through `itemType` property. This should also be extended to cover `type` as well